### ISSUE TITLE
[Bug]: extractinator accepts non-block items but doesn't process them

### ISSUE THREAD

**Issue by deli73** [AlexNijjar/The-Extractinator#41]
### Bug Description

when using a non-block item (e.g. arrows) as input for an extractinator recipe, the items can be placed (or piped via hopper) into the extractinator, but will not be processed and instead simply clog the machine entirely.

### How to Reproduce?

add the following extractinator recipe to a datapack:
```
{
  "type": "extractinator:extractinating",
  "input": {
    "item": "minecraft:arrow"
  },
  "drops": [
    {
      "drop": "minecraft:flint",
      "drop_chance": 0.15
    },
    {
      "drop": "minecraft:stick",
      "drop_chance": 0.2
    }
  ]
}
```
attempt to use the recipe with an extractinator.

### Expected Behavior

the extractinator should process the item as it does any block item, and output the relevant resources.

### Version

2.1.0

### Mod Loader Version

1.19.4 - 0.14.22

### Mod Loader

Fabric

### Logs or additional context

full modlist:
```
	- accurateblockplacement 1.0.17
	- anarchronism 0.6.0
	- antibat 1.19.2
	- areas 5.0
	- axesareweapons 1.7.1
	   \-- com_github_llamalad7_mixinextras 0.2.0-beta.9
	- betterdungeons 1.19.4-Fabric-3.4.0
	   \-- org_reflections_reflections 0.10.2
	- betterf3 6.0.2
	- bettermineshafts 1.19.4-Fabric-3.4.0
	- betterstrongholds 1.19.4-Fabric-3.4.0
	- betterthirdperson 1.9.0
	- bridgingmod 1.19.4-v2.1.0
	- cakechomps 6.0.1+1.19.4
	- cardinal-components 5.1.0
	   |-- cardinal-components-base 5.1.0
	   |-- cardinal-components-block 5.1.0
	   |-- cardinal-components-chunk 5.1.0
	   |-- cardinal-components-entity 5.1.0
	   |-- cardinal-components-item 5.1.0
	   |-- cardinal-components-level 5.1.0
	   |-- cardinal-components-scoreboard 5.1.0
	   \-- cardinal-components-world 5.1.0
	- carpet 1.4.101+v230319
	- citresewn 1.1.3+1.19.4
	   \-- citresewn-defaults 1.1.3+1.19.4
	- cloth-config 10.1.105
	   \-- cloth-basic-math 0.6.1
	- collective 6.65
	- connectivity 1.19.4-4.2
	- continuity 3.0.0-beta.3+1.19.4
	- cooldown-coordinator 0.6.2
	- debugify 1.19.4+2.2
	- ears 1.4.6
	- easyanvils 6.0.0
	- editsign 2.8.0
	- emi 1.0.19+1.19.4+fabric
	- entityculling 1.6.2-mc1.19.4
	- extractinator 2.1.0
	- fabric-api 0.87.0+1.19.4
	   |-- fabric-api-base 0.4.28+737a6ee8f4
	   |-- fabric-api-lookup-api-v1 1.6.29+504944c8f4
	   |-- fabric-biome-api-v1 13.0.9+10ce000ff4
	   |-- fabric-block-api-v1 1.0.8+e022e5d1f4
	   |-- fabric-blockrenderlayer-v1 1.1.38+10ce000ff4
	   |-- fabric-client-tags-api-v1 1.1.1+0eee1254f4
	   |-- fabric-command-api-v1 1.2.31+f71b366ff4
	   |-- fabric-command-api-v2 2.2.10+10ce000ff4
	   |-- fabric-commands-v0 0.2.48+df3654b3f4
	   |-- fabric-containers-v0 0.1.59+df3654b3f4
	   |-- fabric-content-registries-v0 3.5.14+10ce000ff4
	   |-- fabric-convention-tags-v1 1.5.3+6d727b51f4
	   |-- fabric-crash-report-info-v1 0.2.17+aeb40ebef4
	   |-- fabric-data-generation-api-v1 11.5.2+26f09d40f4
	   |-- fabric-dimensions-v1 2.1.50+44c9c860f4
	   |-- fabric-entity-events-v1 1.5.17+10ce000ff4
	   |-- fabric-events-interaction-v0 0.6.1+4b6b93f0f4
	   |-- fabric-events-lifecycle-v0 0.2.56+df3654b3f4
	   |-- fabric-game-rule-api-v1 1.0.35+a1ccd7bff4
	   |-- fabric-item-api-v1 2.1.21+10ce000ff4
	   |-- fabric-item-group-api-v1 3.0.10+8400c67ef4
	   |-- fabric-key-binding-api-v1 1.0.35+504944c8f4
	   |-- fabric-keybindings-v0 0.2.33+df3654b3f4
	   |-- fabric-lifecycle-events-v1 2.2.19+10ce000ff4
	   |-- fabric-loot-api-v2 1.1.32+10ce000ff4
	   |-- fabric-loot-tables-v1 1.1.36+9e7660c6f4
	   |-- fabric-message-api-v1 5.1.5+53c11dadf4
	   |-- fabric-mining-level-api-v1 2.1.44+10ce000ff4
	   |-- fabric-model-loading-api-v1 1.0.2+3a1ea2f1f4
	   |-- fabric-models-v0 0.4.1+a0255436f4
	   |-- fabric-networking-api-v1 1.3.6+ae9c4c6af4
	   |-- fabric-networking-v0 0.3.46+df3654b3f4
	   |-- fabric-object-builder-api-v1 7.1.1+d63b52eaf4
	   |-- fabric-particles-v1 1.1.1+86d48884f4
	   |-- fabric-recipe-api-v1 1.0.13+10ce000ff4
	   |-- fabric-registry-sync-v0 2.3.2+95ae8716f4
	   |-- fabric-renderer-api-v1 2.4.2+90110d8df4
	   |-- fabric-renderer-indigo 1.4.2+90110d8df4
	   |-- fabric-renderer-registries-v1 3.2.42+df3654b3f4
	   |-- fabric-rendering-data-attachment-v1 0.3.32+10ce000ff4
	   |-- fabric-rendering-fluids-v1 3.0.25+10ce000ff4
	   |-- fabric-rendering-v0 1.1.45+df3654b3f4
	   |-- fabric-rendering-v1 2.1.5+10ce000ff4
	   |-- fabric-resource-conditions-api-v1 2.3.3+10ce000ff4
	   |-- fabric-resource-loader-v0 0.11.7+8400c67ef4
	   |-- fabric-screen-api-v1 1.0.49+10ce000ff4
	   |-- fabric-screen-handler-api-v1 1.3.25+10ce000ff4
	   |-- fabric-sound-api-v1 1.0.11+10ce000ff4
	   |-- fabric-transfer-api-v1 3.3.0+7bf81db9f4
	   \-- fabric-transitive-access-wideners-v1 3.3.0+1b5f819af4
	- fabrication 3.3.1+1.19.4
	   \-- lib39-deferral 1.5.0-pre2.2+1.19.3
	- fabricloader 0.14.22
	- fabrictailor 2.1.1
	   |-- config2brigadier 1.2.3
	   \-- fabric-permissions-api-v0 0.2-SNAPSHOT
	- ferritecore 5.2.0
	- forager_bow_ench 1.1.0
	- forgeconfigapiport 6.0.2
	   |-- com_electronwill_night-config_core 3.6.5
	   \-- com_electronwill_night-config_toml 3.6.5
	- grapplemod 1.19.4-v1.7
	- immediatelyfast 1.1.28+1.19.4
	   |-- com_github_llamalad7_mixinextras 0.2.0-beta.9
	   \-- net_lenni0451_reflect 1.2.1
	- indium 1.0.19+mc1.19.4
	- infinitywaterbucket 2.0.0
	- inventory_control_tweaks 1.3.26
	   |-- com_github_llamalad7_mixinextras 0.2.0-beta.9
	   \-- crowdin-translate 1.4+1.19.3
	- itemswapper 0.4.0-mc1.19.4
	- java 17
	- keepmysoiltilled 2.0
	- krypton 0.2.2
	   \-- com_velocitypowered_velocity-native 3.2.0-SNAPSHOT
	- languagereload 1.5.8+1.19.4
	- linearxp 1.0.1
	- lithium 0.11.1
	- logicates 1.1.0+1.19.4
	- lootr 0.6.28.69
	- malilib 0.15.4
	- map_atlases 2.4.0
	- matchbox 0.3.0-1.19.4
	- memoryleakfix 1.1.1
	- midnightlib 1.3.0
	- minecraft 1.19.4
	- modern_beta 6.4.1+1.19.4
	- modernfix 5.7.1+mc1.19.4
	   \-- com_github_llamalad7_mixinextras 0.2.0-beta.9
	- modmenu 6.3.1
	- mousetweaks 2.24
	- nohunger 1.1.0
	- notenoughanimations 1.6.4
	- omnihopper 2.0.1+1.19.4
	   \-- cicada 0.4.0
	- origins 1.9.0
	   |-- apoli 2.8.0
	   |    |-- additionalentityattributes 1.2.2+1.19.4
	   |    |-- calio 1.10.0
	   |    |-- cardinal-components-base 5.1.0
	   |    |-- cardinal-components-entity 5.1.0
	   |    \-- playerabilitylib 1.7.0
	   \-- reach-entity-attributes 2.3.2
	- overflowingbars 6.0.0
	- owo 0.10.6+1.19.4
	   \-- blue_endless_jankson 1.2.2
	- pehkui 3.7.8+1.14.4-1.20.1
	   \-- kanos_config 0.4.1+1.14.4-1.19.4
	- pettable 1.3.2
	- ping-wheel 1.4.0
	- puzzleslib 6.0.11
	- quickshulker 1.4.0-1.19.4
	   |-- kyrptconfig 1.5.3-1.19.4
	   \-- shulkerutils 1.0.4-1.19
	- raknetify 0.1.0+alpha.5.97
	- repurposed_structures 6.3.25+1.19.4-fabric
	- resourcefulconfig 1.2.0
	- resourcefullib 1.3.6
	   \-- com_teamresourceful_yabn 1.0.3
	- rrls 1.19.4-3.1.0-fabric
	- scaffoldingdropsnearby 3.0
	- server-hats 0.15
	- shimmer 1.19.4-0.1.15
	- shulkerboxtooltip 3.4.0+1.19.4
	- simplemagnets 1.1.9
	- smoothboot 1.19.4-1.7.0
	- sodium 0.4.10+build.24
	- sound_physics_remastered 1.19.4-1.1.1
	- spyglass-improvements 1.4+mc1.19.4+fabric
	- starlight 1.1.1+fabric.ae22326
	- styled-nicknames 1.2.0+1.19.3
	   \-- player-data-api 0.2.2+1.19.3
	- styledchat 2.1.6+1.19.4
	   |-- fabric-permissions-api-v0 0.2-SNAPSHOT
	   |-- placeholder-api 2.1.0+1.19.4
	   |-- player-data-api 0.2.2+1.19.3
	   \-- predicate-api 0.1.0+1.19.3
	- suitablystackablestew 1.1.2
	- supermartijn642configlib 1.1.8
	- supermartijn642corelib 1.1.12+a
	- switchy 2.7.3+1.19.4
	   |-- switchy-cardinal 2.7.3+1.19.4
	   |-- switchy-cardinal-ui 2.7.3+1.19.4
	   |-- switchy-client 2.7.3+1.19.4
	   |-- switchy-compat 2.7.3+1.19.4
	   |-- switchy-compat-ui 2.7.3+1.19.4
	   |-- switchy-core 2.7.3+1.19.4
	   |    \-- server_translations_api 2.0.0-beta.2+1.19.4-pre2
	   |         \-- packet_tweaker 0.3.1+23w06a
	   \-- switchy-ui 2.7.3+1.19.4
	- tinkerers_smithing 2.3.1+1.19.4
	- tooltipfix 1.1.1-1.19.3
	- travelersbackpack 1.19.4-8.4.8
	   \-- reach-entity-attributes 2.3.2
	- tweakeroo 0.16.0
	- waterdripsound 1.19-0.3.2
	- whereisit 1.14.16
	- worldedit 7.2.14+6360-094b734
	- worldeditcui 1.19.4+01
	- yet_another_config_lib_v3 3.1.0+1.19.4
	   |-- com_twelvemonkeys_common_common-image 3.10.0-SNAPSHOT
	   |-- com_twelvemonkeys_common_common-io 3.10.0-SNAPSHOT
	   |-- com_twelvemonkeys_common_common-lang 3.10.0-SNAPSHOT
	   |-- com_twelvemonkeys_imageio_imageio-core 3.10.0-SNAPSHOT
	   |-- com_twelvemonkeys_imageio_imageio-metadata 3.10.0-SNAPSHOT
	   |-- com_twelvemonkeys_imageio_imageio-webp 3.10.0-SNAPSHOT
	   \-- yet_another_config_lib 2.5.1+1.19.4
	- yungsapi 1.19.4-Fabric-3.10.1
	   \-- org_javassist_javassist 3.29.2-GA
	- yungsextras 1.19.4-Fabric-3.3.0
```

### Code of Conduct

- [X] I have searched the issue tracker and confirmed that the issue has not been reported
- [X] I have checked the FAQ (if one is present) and my issue is not listed
- [X] I have verified that I am using the latest version of the mod
- [x] I have verified that I do not have Optifine installed. We will close issues if we detect Optifine is in your pack. We cannot fix Optifine issues. Please do not report issues with Optifine present.

**Comment by Darkosto** [comment]
I'm also getting the same behavior on the modpack I'm working on currently. Is there any chance this will be looked at for Minecraft 1.20.1? Thanks!

**Comment by AlexNijjar** [comment]
Was originally planning on fixing this in 1.20.3. However, if there’s demand for 1.20.1, I’m happy to backport as well.

**Comment by deli73** [comment]
any chance a fix for 1.19.4 would be in the cards too?

**Comment by Darkosto** [comment]
A 1.20.1 version would be much appreciated @AlexNijjar 

Thank you!

**Comment by AlexNijjar** [comment]
> any chance a fix for 1.19.4 would be in the cards too?

Sure.
